### PR TITLE
CMake: tests-lit target for running LIT tests

### PR DESCRIPTION
**Pull request by stanislaw**

### CODE REVIEW

File: tests-lit/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+add_custom_target(tests-lit)
+
+set(CLANG_EXEC ${PATH_TO_LLVM}/bin/clang++)
+set(FILECHECK_EXEC filecheck)
```
Comment (stanislaw) [inline tests-lit/CMakeLists.txt]: What is the best way to check / ensure that `filecheck` exists and throw an error otherwise?

Comment (AlexDenisov) [reply tests-lit/CMakeLists.txt]: I think the 'right' way is to use [find_program](https://cmake.org/cmake/help/latest/command/find_program.html), but it's up to you.

File: tests-lit/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+add_custom_target(tests-lit)
```
Comment (AlexDenisov) [inline tests-lit/CMakeLists.txt]: What does it do?

Comment (stanislaw) [reply tests-lit/CMakeLists.txt]: It is a target that simply executes the custom command above. Do you see anything wrong with this approach?

Comment (AlexDenisov) [reply tests-lit/CMakeLists.txt]: Not necessarily wrong, but looks a bit suspicious to me, though it's just a gut feeling. I think it's fine.

File: tests-lit/CMakeLists.txt (hunk 1 above)
Comment (stanislaw) [reply tests-lit/CMakeLists.txt]: As discussed in person, we leave it with `set`.

File: tests-lit/CMakeLists.txt (hunk 2 above)
Comment (stanislaw) [reply tests-lit/CMakeLists.txt]: This has been changed.

**Pull request merged**